docs/content/concepts/websites.md

```diff
@@ -0,0 +1,113 @@
++++
+title="Websites"
+weight=4
+[taxonomies]
+tags = ["concepts"]
++++
+
+Boltzmann can be configured with features that give you a jumpstart with
+building websites. Specifically, you can build services that respond to route
+handlers with templated text/html responses as well as JSON responses.
+
+<!-- more -->
+
+Enable the built-in cross-site request forgery protection by passing `--csrf`.
+Enable the built-in template rendering by passing `--templates`. Cookie features
+are built into Boltzmann.
+
+To scaffold a service with all website-focused features, pass the `--website` option.
+
+
+```shell
+> boltzmann-cli --website --silent my-website
+Boltzmann at 0.1.2 with csrf, githubci, jwt, ping, status, templates
+```
+
+Run the server with `node my-website/boltzmann.js`. Point your browser to
+`http://localhost:5000/hello/world` to see the simple example template being rendered.
+
+## Templates
+
+Boltzmann's templates feature uses
+[Nunjucks](https://mozilla.github.io/nunjucks/) templates. Template files by
+default live in the `./templates` directory. Scaffolding with this feature
+creates templates directory with an example `index.html` file for you, along
+with a handler that renders it.
+
+To respond from a route with a rendered template, name the template file in the
+object return by the route handler, like this:
+
+```
+return {
+  [Symbol.for('template')]: 'index.html',
+  cats: ['Fezzik', 'Mina', 'Oswin' ],
+  count: 10
+}
+```
+
+Everything else in the object is passed to the template renderer as context.
+
+Boltzmann looks for error templates in the `./templates` directory. It
+follows the convention of looking for `4xx.html` for all 400-category errors,
+and `5xx.html` for all 500-category errors.
```

Comment:
We might expand here (or change our behavior): we use `4xx` and `5xx` for _thrown_ errors that do not have a template otherwise specified. If you _return_ a `404` or `503` without a template specified, those will be rendered as JSON.

Reply:
(Also, in dev mode, we don't use the app's 4xx/5xx templates unless specifically requested from the dev error page.)